api/controllers/user.go

```diff
@@ -85,6 +85,17 @@ func (uc *UserController) Login(w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(201)
 }
 
+// @Summary Update user information
+// @Description Updates the details of an existing user based on the provided user ID and payload
+// @Tags User
+// @Produce json
+// @Param userId path string true "User ID"
+// @Param user body dto.UserUpdateDto true "User update payload"
+// @Success 200 {object} dto.UserUpdateDto
+// @Failure 400
+// @Failure 404
+// @Failure 500
+// @Router /users/{userId} [put]
 func (uc *UserController) Update(w http.ResponseWriter, r *http.Request) {
 	IDStr := chi.URLParam(r, "id")
```

Comment:
Note que tanto a rota r.Patch("/{id}", contro...) está usando id, então você precisa usar o mesmo parâmetro na notação do swagger (// @Param id path string true "User ID"). Sugiro alterar todos para userID (para padronizar ao invés de id ou userId). Acredito que por isso não está funcional a sua rota. Ela deveria parecer com isso aqui 

![Captura de tela de 2024-09-11 22-27-28](https://github.com/user-attachments/assets/72791658-9113-45c2-95bb-52fec9767c25)

Note que ao clicar em try in out foi liberado locais para adicionar ID e um body para atualizar.
OBS: sempre que alterar as anotações swag, gere uma nova doc com o comando make swag.